### ISSUE TITLE
Kernel: x86_64 ap_setup.S is non-functional

### ISSUE THREAD

**Issue by ADKaster** [SerenityOS/serenity#12662]
For SMP, we need to be able to boot secondary CPUs (AP, Application Processors) from the primary CPU (BSP, BootStrap Processor).

For i686 and x86_64, This happens in `Kernel::APIC::setup_ap_boot_environment()` and `Kernel::APIC::do_boot_aps()`.

setup_ap_boot_environment() sets up a code segment, initial stack, and MSR values for each AP. do_boot_aps() creates an idle thread for them, and sends an initial Interprocessor Interrupt (IPI) to each one to perform its initial boot and eventually start executing the idle thread.

For x86_64, the "initial boot" step is broken. The most obvious broken part is the global descriptor table pointer, which is commented out completely. There's a global variable in both the Prekernel/boot.S and in Arch/x86/x86_64/Boot/ap_setup.S that is supposed to hold a pointer to the global descriptor table for use in long mode. However in ap_setup.S, it's commented out, presumably because the way the assembly is currently written it'll try to create an R_X86_64_32 relocation against `gdt64ptr` when loading the GDT from that address.

The value of `gdt64ptr` allegedly comes from the pre-kernel, however, in Prekernel/boot.S, the value there is a 10 byte value, and the value passed from the prekernel to the kernel is only 4 bytes. The [Intel® 64 and IA-32 Architectures Software Developer’s Manual Volume 2A: Instruction Set Reference, A-L](https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-2a-manual.pdf) page 632 suggests that it's really supposed to be a 10 byte value for the final long mode GDT value.

So, long story short:

1) SMP boot in x86_64 is broken
2) It's related to loading  intermediate global descriptor tables getting APs from protected mode to long mode
3) the gdt64ptr being passed from the prekernel to the kernel is suspicious.
